onnx/checker.cc

```diff
@@ -35,15 +43,138 @@ namespace checker {
     }                                         \
   } while (0)
 
-void check_value_info(const ValueInfoProto& value_info, const CheckerContext&) {
+bool check_name_syntax(const std::string& name) {
+  if (name.empty()) {
+    return false;
+  }
+
+  bool correct = true;
+
+  // Names should adhere to C identifier syntax.
+  auto iter = name.cbegin();
+
+  char c = *iter;
+  if (!(isalpha(c) || c == '_')) {
+    correct = false;
+  }
+
+  ++iter;
+
+  for (; iter < name.cend(); ++iter) {
+    c = *iter;
+    if (!(isalnum(c) || c == '_')) {
+      correct = false;
+    }
+  }
+
+  return correct;
+}
+
+bool check_domain_syntax(const std::string& name) {
+  if (name.empty()) {
+    return false;
```

Comment:
if domain is empty, it's default domain. We should accept it.

Reply:
True for nodes, and possibly true for models, depending on resolution of #1222. The check was intended for models, so I'm adding a separate check for nodes that accept an empty domain.